pkg/controller/network/publicipaddress/managed.go

```diff
@@ -98,12 +101,17 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(azureclients.IsNotFound, err), errGetPublicIPAddress)
 	}
 
+	network.LateInitializePublicIPAddress(&s.Spec.ForProvider, &az)
+	if err := e.kube.Update(ctx, s); err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errUpdateCR)
+	}
+
```

Comment:
We usually update only if there is a diff and used to do `kube.Update` for late-init in old controllers but the recommended way today is to return `ResourceLateInitialized: true` to make managed reconciler handle it.

Reply:
Thanks @muvaf for the `ResourceLateInitialized: true` suggestion. I've removed the embedded controller-runtime client from our external client implementation and now always set `ObservationResult.ResourceLateInitialized` accordingly. 

We can at a later time revisit `provider-azure`'s late-initialization utility functions so that they report whether fields have been late-initialized. 

While working on this, I had also considered using `cmp.Equal` to detect any changes after late-initialization, however, I did not choose to do so as API server already detects noops and does not update the resource. However, leaving update calls to the managed reconciler is I think a good compromise.

Reply:
I just realized that if we leave this to the managed reconciler, we are losing any status updates. So, I'm reverting back to making the update call in-site as we already do in the repo:
https://github.com/crossplane/provider-azure/blob/8380dd2d85e0391bd02e23c5c47c13f8962c38a7/pkg/controller/database/postgresqlserver/managed.go#L122

So returning `ResourceLateInitialized: true` makes us lose status updates and I'd like to address proper calculation of this in a repo-wide manner.